src/admin/components/forms/field-types/Array/index.tsx

```diff
@@ -41,6 +42,7 @@ const ArrayFieldType: React.FC<Props> = (props) => {
     admin: {
       readOnly,
       description,
+      header,
```

Comment:
I'd put this on the `admin.components` property and name it `RowHeader` to be as explicit as possible. That way if we ever allow for a custom `Header` component or similar, say maybe used in the Edit view, we would have no overlap between the two. 

It also makes sense to me to just have this component _**always**_ be a React component. Might simplify the work necessary a bit.

Also note that if a config property corresponds to a React component, it should be title case to match React conventions.

Reply:
A yes, I didn't see the `admin.components` yet, adding the prop there does make semantically sense.
So, just to be 100% sure, the config would become something like `{type: "array", components: {RowHeader: ... }}`?